api/src/main/java/org/pmiops/workbench/api/DataBrowserController.java

```diff
@@ -0,0 +1,292 @@
+                            .analysisId( cdr.getAnalysisId())
+                            .analysisName(cdr.getAnalysisName())
+                            .stratum1Name(cdr.getStratum1Name())
+                            .stratum2Name(cdr.getStratum2Name())
+                            .stratum3Name(cdr.getStratum3Name())
+                            .stratum4Name(cdr.getStratum4Name())
+                            .stratum5Name(cdr.getStratum5Name())
+                            .chartType(cdr.getChartType())
+                            .dataType(cdr.getDataType());
+
+                }
+            };
+
+    /**
+     * Converter function from backend representation (used with Hibernate) to
+     * client representation (generated by Swagger).
+     */
+    private static final Function<DbDomain, org.pmiops.workbench.model.DbDomain>
+            TO_CLIENT_DBDOMAIN =
+            new Function<DbDomain, org.pmiops.workbench.model.DbDomain>() {
+                @Override
+                public org.pmiops.workbench.model.DbDomain apply(org.pmiops.workbench.cdr.model.DbDomain cdr) {
+                    return new org.pmiops.workbench.model.DbDomain()
+                            .domainId( cdr.getDomainId())
+                            .domainDisplay(cdr.getDomainDisplay())
+                            .domainDesc(cdr.getDomainDesc())
+                            .domainParent(cdr.getDomainParent())
+                            .domainRoute(cdr.getDomainRoute());
+
+
+                }
+            };
+
+
+
+    /**
+     * This method searches concepts
+     *
+     * @param conceptName
+     * @param standardConcept
+     * @param concept_code
+     * @param vocabulary_id
+     * @param domain_id
+     * @return
+     */
+    @Override
+    public ResponseEntity<ConceptListResponse> getConceptsSearch(
+            String conceptName,
+            String standardConcept,
+            String concept_code,
+            String vocabulary_id,
+            String domain_id) {
+
+
+        List<Concept> conceptList;
+
+        // If Concept name do search on name
+
+        if (conceptName != null) {
+            conceptList = conceptDao.findConceptLikeName(conceptName);
+        }
+        else {
+            conceptList = conceptDao.findConceptsOrderedByCount();
+        }
+
+        ConceptListResponse resp = new ConceptListResponse();
+        resp.setItems(conceptList.stream().map(TO_CLIENT_CONCEPT).collect(Collectors.toList()));
+
+        return ResponseEntity.ok(resp);
+    }
+
+    /**
+     * This method gets concepts with maps to relationship in concept relationship table
+     *
+     * @param conceptId
+     * @param conceptNum
+     * @return
+     */
+    @Override
+    public ResponseEntity<ConceptListResponse> getConceptsMapsTo(
+            Long conceptId,
+            Long conceptNum) {
+
+
+        List<Concept> conceptList;
+
+        // If Concept num is 2 , children
+        if (conceptNum == 2 ) {
```

Comment:
Should you make a static constant to represent the 2? For example:

`private static final int CONCEPT_CHILDREN = 2;`

then you can reference it like so:

`if (conceptNum == CONCEPT_CHILDREN) {`

Reply:
You can also define an enum in workbench.yaml and reference it here. That way the client and server share the enum.

Reply:
Not my style, but i can if you like. Want me to?  The conceptNum refers to the column name -- either concept_id_1 or concept_id_2 .  This is only place it is used.

Reply:
Normally speaking we try to define REST APIs that people other than the author can understand. Using enums when appropriate helps with that. So personally I would.

If what you're looking for are parents or children, ideally the API should ask for *that* (either via an enum parameter or different endpoints), not pass up a "1" or a "2".

But you certainly can accept a number here instead, though defining constants for the numbers somewhere as Brian suggests may be a good idea. And if you're going to do this, you should do your own validation of the input (that you only get 1 or 2, and not 27 or whatever.)